formula-coroutines/src/main/java/com/instacart/formula/coroutines/FlowStream.kt

```diff
@@ -0,0 +1,82 @@
+package com.instacart.formula.coroutines
+
+import com.instacart.formula.Cancelable
+import com.instacart.formula.Stream
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.runBlocking
+
+/**
+ * Formula [Stream] adapter to enable coroutine's Flow use.
+ */
+interface FlowStream<Message> : Stream<Message> {
+
+    companion object {
+        /**
+         * Creates a [Stream] from a [Flow] factory [create].
+         *
+         * ```
+         * events(FlowStream.fromFlow { locationManager.updates() }) { event ->
+         *   transition()
+         * }
+         * ```
+         */
+        inline fun <Message> fromFlow(
+            scope: CoroutineScope,
+            crossinline create: () -> Flow<Message>
+        ): Stream<Message> {
+            return object : FlowStream<Message> {
+
+                override val scope: CoroutineScope = scope
+
+                override fun flow(): Flow<Message> {
+                    return create()
+                }
+
+                override fun key(): Any = Unit
+            }
+        }
+
+        /**
+         * Creates a [Stream] from a [Flow] factory [create].
+         *
+         * ```
+         * events(FlowStream.fromFlow(itemId) { repo.fetchItem(itemId) }) { event ->
+         *   transition()
+         * }
+         * ```
+         *
+         * @param key Used to distinguish this [Stream] from other streams.
+         */
+        inline fun <Message> fromFlow(
+            scope: CoroutineScope,
```

Comment:
Could we provide a default global scope here? The lifecycle of `Stream` will be controlled by `Formula` so in most cases, we don't really need to control it using scope.

Reply:
I thought about this. Using coroutine's own `GlobalScope` is mainly considered bad practice. I am just not sure how to create a `Scope` for a formula since its lifecycle is really controlled by Android. For my example `StopwatchFormula` uses `MainScope` which uses a `SupervisorJob` + `Dispatchers.Main`. I could move this default into `fromFlow`.

Reply:
`MainScope` seems like a decent default.